lib/Test/MockFile.pm

```diff
@@ -307,17 +466,74 @@ sub _strict_mode_violation {
     my @stack = _get_stack();
     return unless scalar @stack;    # skip the package
 
+    my $filename;
+
     # check it later so we give priority to authorized_strict_mode_packages
     my $file_arg = file_arg_position_for_command( $command, $at_under_ref );
 
-    my $filename = scalar @$at_under_ref <= $file_arg ? '<not specified>' : $at_under_ref->[$file_arg];
+    if ( $file_arg >= 0 ) {
+        $filename = scalar @$at_under_ref <= $file_arg ? '<not specified>' : $at_under_ref->[$file_arg];
+    }
 
     # Ignore stats on STDIN, STDOUT, STDERR
-    return if $filename =~ m/^\*?(?:main::)?[<*&+>]*STD(?:OUT|IN|ERR)$/;
+    return if defined $filename && $filename =~ m/^\*?(?:main::)?[<*&+>]*STD(?:OUT|IN|ERR)$/;
+
+    my $path = _abs_path_to_file($filename);
+
+    my $context = {
+        command      => $command,
+        filename     => $path,
+        at_under_ref => $at_under_ref
+    };    # object
+
+    return if _dispatch_strict_rules($context);
+
+    croak("Unknown strict mode violation for $command") if $file_arg == -1;
 
     confess("Use of $command to access unmocked file or directory '$filename' in strict mode at $stack[1] line $stack[2]");
 }
 
+sub _dispatch_strict_rules {
+    my ($context) = @_;
+
+    # rules dispatch
+    foreach my $rule (@STRICT_RULES) {
+
+        # validate file_rule
+        next unless _check_rule_condition->( $context->{filename}, $rule->{file_rule} );
```

Comment:
I think it's better to send the filename and command to one check, because then it's:

```perl
foreach my $rule (@rules) {
    $rule->{'filename'} eq '*' ||  first( sub { $filename =~ /\Q$_\E/xms }, $rule->{'files'}->@* )
        or next; # try the next rule

     $rule->{'command'} eq '*' || first( sub { $command =~ /\Q$_\E/xms }, $rule->{'commands'}->@* )
        or next;

    return $rule->{'action'}->( $command, $filename, $ctx );
}

return 1; # couldn't find any matches, just return success
```